gce/burst-training/README.md

```diff
@@ -0,0 +1,327 @@
+
+
+### Training code
+
+[census-analysis.py](./census-analysis.py) contains the training code we will
+run for the Census problem. The only important thing about this code, as far as
+adapting it to your own use case, is the command line interface. This is the
+means by which we will get our burst training instance to pass parameters to the
+training job.
+
+In the case of the Census code, we have to specify three parameters when we run
+the job:
+
+1. `--census-data-path` -- the Cloud Storage path to the training and test data
+
+2. `--model-output-path` -- the Cloud Storage path at which the trainer should
+   store its trained model binary
+
+3. `--cv-iterations` -- the number of hyperparameter configurations that
+   `RandomizedSearchCV` should try
+
+That particular trainer also allows you to specify a `--mode`, which defaults
+to `train`. This `--mode` argument will be useful to us once the training job
+has completed, as it will allow us to test out the trained model from our local
+environment.
+
+*Note: The [census-analysis.py](./census-analysis.py) trainer uses TensorFlow's
+[tf.gfile](https://www.tensorflow.org/api_docs/python/tf/gfile) to store and
+retrieve data from Cloud Storage. This is the only capacity in which it uses
+TensorFlow. This is also a very useful module to know about -- it allows you to
+interact with Cloud Storage objects through your Python environment just as you
+would interact with files on your own filesystem.*
+
+
+### Cloud Storage bucket
+
+Before execution, you will have to place the training and evaluation data on
+[Google Cloud Storage](https://cloud.google.com/storage/docs/).
+
+The executor for the Census example expects a Cloud Storage Bucket of your
+choice with the data stored under the path `gs://<BUCKET-NAME>/census/`. You can
+[download the data from the UCI Machine Learning
+Repository](https://archive.ics.uci.edu/ml/machine-learning-databases/adult/) -
+the `adult.data` and `adult.test` files are all you need. Once you have them
+available to you locally, you can copy them over to the desured Cloud Storage
+path by running the following command from the directory containing those files:
+
+```
+gsutil -m cp adult.* gs://<BUCKET-NAME>/census/
+```
+
+You may have to create the bucket first - either using `gsutil mb
+gs://<BUCKET-NAME>` or through the [Cloud Console Storage
+browser](https://console.cloud.google.com/storage/browser).
+
+Alternatively, you can also use the [Storage
+browser](https://console.cloud.google.com/storage/browser) in the Cloud Console.
+
+The executor, which we cover in more detail below, will:
+
+1. Upload the training script to `gs://<BUCKET-NAME>`
+
+2. Run the training script on the data in `gs://<BUCKET-NAME>/census`
+
+3. Save the trained model to `gs://<BUCKET-NAME>/census-<TIMESTAMP>`, with the
+   `<TIMESTAMP>` representing the time at which the job was submitted
+
+**Please replace <BUCKET-NAME> with your own name.**
+
+
+### Execution
```

Comment:
Add subsections in execution:

1. Setup
2. Train.sh explanation (uses census-startup.sh and census-analysis.py)
3. Checklist

Explain how these files get to GCE instance.

Reply:
Done